File: clients/swift/Sources/Komus24/Model/Order.swift
/// Player's orders
public struct Order {
    /// TODO - Document
    let vehicles: [VehicleOrder]

    /// Read Order from input stream
    static func readFrom<S: InputStream>(_ stream: S) -> Order {
        var vehicles: [VehicleOrder]
        let vehiclesSize = stream.readInt32()
        vehicles = (0..<vehiclesSize).map{ _ in
            var vehiclesSize: VehicleOrder
            vehiclesSize = VehicleOrder.readFrom(stream)
            return vehiclesSize
        }
        return Order(vehicles: vehicles)
    }

    /// Write Order to output stream
    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeInt32(Int32(vehicles.count))
        for vehiclesElement in vehicles {
            vehiclesElement.writeTo(stream)
        }
    }
}

File: clients/scala/src/main/scala/komus24/model/Order.scala
package komus24.model

import komus24.util.StreamUtil

/**
 * Player's orders
 *
 * @param vehicles TODO - Document
 */
case class Order(vehicles: Seq[komus24.model.VehicleOrder]) {
    /**
     * Write Order to output stream
     */
    def writeTo(stream: java.io.OutputStream): scala.Unit = {
        StreamUtil.writeInt(stream, vehicles.length)
        vehicles.foreach { value =>
            value.writeTo(stream)
        }
    }

    /**
     * Get string representation of Order
     */
    override def toString(): String = {
        var stringBuilder = new StringBuilder("Order { ")
        stringBuilder.append("vehicles: ")
        stringBuilder.append(vehicles)
        stringBuilder.append(" }")
        stringBuilder.toString()
    }
}

object Order {
    /**
     * Read Order from input stream
     */
    def readFrom(stream: java.io.InputStream): Order = Order(
        (0 until StreamUtil.readInt(stream)).map { _ =>
            komus24.model.VehicleOrder.readFrom(stream)
        }
    )
}